telemetry/telemetry.proto

```diff
@@ -4,6 +4,57 @@ import 'nanopb.proto';
 
 option java_outer_classname="TelemetryProto";
 
-message Telemetry {
-  optional int32 timestamp_ms = 1;
+//Should be sent only after device resets
+message NameIDMessage { //Send the device ID with its corresponding name
+	optional uint32 id = 1;
+	optional string name = 2 [(nanopb).max_size = 32];
+}
+
+message StatisticalAggregate {
+  optional uint32 numSamples = 1;
+  optional uint32 min = 2;
+  optional uint32 max = 3;
+  optional uint32 mean = 4;
+  optional uint32 stdev = 5;
+}
+
+message GeneralReading { //this might want a more specific name?
+	enum READING_TYPE {
+		VOLTAGE = 1;
+		CURRENT = 2;
+		TEMPERATURE = 3;
+	}
+	optional READING_TYPE type = 1;
+	optional uint32 id = 2; //id of the device being read
+	optional StatisticalAggregate reading = 3;
+}
+
+message MCCInfoMessage {
+	optional StatisticalAggregate speed = 1; //information about the entire car's speed
+}
+
+message PedalMessage {
+	optional StatisticalAggregate accelPos = 1;
+	optional StatisticalAggregate brakePos = 2;
+}
+
+message ErrorMessage {
+	optional uint32 id = 1; //id of the device that's causing the error
```

Comment:
Generally the way I've defined things are (and note that there may be better ways):
- Datalogger itself doesn't look for anomalies in otherwise reported data, that's outside the scope of the system and should be done by datalogger parser. Generally, the philosophy is that datalogger is just a "dumb pipe".
- In some cases, the data reporting format can't report certain anomalies, so these are logged as strings. One example is RTC time, the format doesn't have a field for whether the RTC's hardware oscillator stopped. In that "exceptional" case, it is reported as a string that the RTC may be inaccurate.
- Generally, the string / info message is meant as a fallback for either system messages, or important data not otherwise recorded. It is not meant for routine data, since strings generally can't be automatically processed (unlike structured data) and are space-inefficient.

For the heartbeat example, I would be tempted to have a message reporting heartbeats, or rather the period between heartbeats as a statistical measurement. This could also generalize into interval between certain types of messages (for example, it would be nice to know that dashboard is sending pedal messages regularly).
Of course, one trade-off to note is that the telemetry viewer needs metadata to report anomalies (for example, expected heartbeat interval), so there are more distributed sources of metadata that need to be kept updated. Unification of that from a single source may be an eventual goal.